elastic/kibana
# [Response Ops][Task Manager] Setting background task utilization window based on poll interval

## Summary

Setting a desired window length (15 seconds) for the background task utilization window running average and calcuating the number of entries to average based on the poll interval. Updating config to be optional so this value can still be configured but if not configured will be calculated.

## To Verify

- Add a console log to log the window size

```
--- a/x-pack/platform/plugins/shared/task_manager/server/monitoring/background_task_utilization_statistics.ts
+++ b/x-pack/platform/plugins/shared/task_manager/server/monitoring/background_task_utilization_statistics.ts
@@ -59,6 +59,7 @@ export function createBackgroundTaskUtilizationAggregator(
     workerUtilizationRunningAverageWindowSize ??
     WORKER_UTILIZATION_RUNNING_AVERAGE_WINDOW_SIZE_MS / pollInterval;

+  console.log(`workerUtilizationWindowSize ${workerUtilizationWindowSize}`);
   const taskRunEventToAdhocStat = createTaskRunEventToAdhocStat();
   const taskRunAdhocEvents$: Observable<Pick<BackgroundTaskUtilizationStat, 'adhoc'>> =
```

- Start Kibana with `mget` claim strategy. The window size should be 30
- Start Kibana with `update_by_query` claim strategy. The window size should be 5
- Set `xpack.task_manager.worker_utilization_running_average_window: 22` in your Kibana config and start Kibana. The window size should be 22 (takes the configured window size if it's passed in).
